detection_rules/devtools.py

```diff
@@ -635,6 +635,28 @@ def license_check(ctx, ignore_directory):
     ctx.exit(int(failed))
 
 
+@dev_group.command('rule-version-check')
+@click.pass_context
+def rule_version_collision_check(ctx: click.Context):
+    """Simulate the incremental step in the version locking to find version change violations."""
+    invalid = []
+    rc = RuleCollection.default()
+    current_stack_ver = current_stack_version()
+
+    click.echo(f'Checking for version collisions in release {current_stack_ver} in {len(rc)} rules')
+    for rule in rc.rules:
+        version_space = rule.contents.get_version_space()
+        if rule.contents.is_dirty and version_space is not None and version_space < 1:
```

Comment:
Just so I understand correctly, if a rule change is detected but buffer space is 0, it is no longer valid? Should we add a wiki on what to do if a forked rule reaches it's buffer limit? We could point them this documentation on how to continue or do we just say "deprecate rule and create a new one"?

Reply:
good call - I need to add support for capturing `max_version` for previous entries